### ISSUE TITLE
Registering with email that may have been used previously on a different account

### ISSUE THREAD

**Issue by Sam-Amy** [BiologicalRecordsCentre/iRecord#1695]
Registering with email address npms@ceh.ac.uk is causing it to be assigned the warehouse ID of another existing account (16), and automatically changing the email address now associated with that account in the warehouse, with the iRecord People table showing two accounts with ID16.  I assume this may be because the npms email was at some point associated with user ID16, and perhaps needs totally purging from the database (i.e. even with deleted flag), though I'm not sure if/how I can do this. 

I've edited the email address for ID16 in the people table of the warehouse back to what it should be, and deleted the iRecord NPMS acccount for now, but it would be helpful to be able to open a seperate account with this email address if the above can be resolved easily.

**Comment by johnvanbreda** [comment]
This is because when an email address is changed, the old email address is kept in the user_identifiers table. Account 16 does have the npms email in its user identifiers table so not only do you need to fix the email in the people table, you should remove the email from user_identifiers in order to properly remove the email address from account 16.
```sql
-- check
select * from user_identifiers where user_id=16;
-- now delete
delete from user_identifiers where user_id=16 and identifier='npms@ceh.ac.uk'
```

I've not run this script so let me know if you need a hand.

**Comment by Sam-Amy** [comment]
Thanks John. 

It looks like this can't be done via the warehouse UI, in which case @BirenRathod are you able to run the following please as I have read only access? Thanks

```sql
delete from user_identifiers where user_id=16 and identifier='npms@ceh.ac.uk'
```

**Comment by BirenRathod** [comment]
@Sam-Amy you mentioned at the top "I've edited the email address for ID16 in the people table of the warehouse back to what it should be, " What email address have you edited with?

**Comment by Sam-Amy** [comment]
bire@ceh.ac.uk - apologies if this is wrong! You can change here is so: [https://warehouse1.indicia.org.uk/index.php/person/edit_from_user/16](https://warehouse1.indicia.org.uk/index.php/person/edit_from_user/16)

**Comment by BirenRathod** [comment]
@Sam-Amy I don't think that has changed anything? I can still see npms@ceh.ac.uk address in the people table in the database.
![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/5941255/368420bb-999a-425f-866c-9b3a01d41abc)

**Comment by Sam-Amy** [comment]
Oh yes, so can I now - that is strange becasue I did change and save, though this was before I then deleted the NPMS account so we can try again now.

**Comment by Sam-Amy** [comment]
I think it has saved now?

**Comment by BirenRathod** [comment]
@Sam-Amy  It has. Also, I deleted the record from user_identifier now.
Thanks, @johnvanbreda.

**Comment by Sam-Amy** [comment]
Great, thanks @Biren and @johnvanbreda, I have now re-registered and a new user ID has been assigned. Strangely, when I registered I got these contradictory messages, but the email did come through so I don't think a concern:

![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/104146164/8c0c88aa-8cfc-412f-bbdd-4b313920a4b4)

**Comment by johnvanbreda** [comment]
Those messages come from Drupal, not Indicia. Maybe worth taking a look @BirenRathod in case it relates to recent email setup changes?

**Comment by Sam-Amy** [comment]
Thanks John, I'll just re-open this for now rather than start a new issue

**Comment by MarkAlliston** [comment]
Needs further investigation
